File: backend/routes/requests.js
const express = require('express');
const QueuedRequest = require('../models/QueuedRequest');

const router = express.Router();

// Get all pending requests
router.get('/', async (req, res) => {
  try {
    const requests = await QueuedRequest.getAll('pending');
    res.json(requests);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// Get request by ID
router.get('/:id', async (req, res) => {
  try {
    const request = await QueuedRequest.findById(req.params.id);
    if (!request) {
      return res.status(404).json({ error: 'Request not found' });
    }
    res.json(request);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

module.exports = router;

File: backend/routes/requests.py
from flask import Blueprint, request
from models import QueuedRequest

requests_bp = Blueprint('requests', __name__)

@requests_bp.route('/', methods=['GET'])
def get_pending_requests():
    """Get all pending requests"""
    try:
        requests = QueuedRequest.query.filter_by(status='pending').all()
        return [req.to_dict(include_creator=True) for req in requests], 200
    except Exception as e:
        return {'error': str(e)}, 500

@requests_bp.route('/<int:request_id>', methods=['GET'])
def get_request(request_id):
    """Get request by ID"""
    try:
        queued_request = QueuedRequest.query.get(request_id)
        if not queued_request:
            return {'error': 'Request not found'}, 404
        return queued_request.to_dict(include_creator=True), 200
    except Exception as e:
        return {'error': str(e)}, 500
